src/request.rs

```diff
@@ -238,27 +267,176 @@ impl RequestBuilder {
     }
 }
 
+impl MultipartRequestBuilder {
+    /// Constructs a new request.
+    pub fn new(request_builder: RequestBuilder) -> MultipartRequestBuilder {
+        MultipartRequestBuilder { request_builder: request_builder, params: None, files: None }
+    }
+
+    /// Add a `Header` to this Request.
+    ///
+    /// ```rust
+    /// # use reqwest::Error;
+    /// #
+    /// # fn run() -> Result<(), Error> {
+    /// use reqwest::header::UserAgent;
+    /// let client = reqwest::Client::new()?;
+    ///
+    /// let res = client.get("https://www.rust-lang.org")?
+    ///     .header(UserAgent("foo".to_string()))
+    ///     .send()?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn header<H>(&mut self, header: H) -> &mut MultipartRequestBuilder
```

Comment:
Well, you'd still be able to call the `json`, `form` and others like the `post` method if you want to keep things in the regular builder. I could potentially fix that by emptying the body in the `MultipartRequestBuilder.build` method and then writing everything to it, but it just seems less elegant to me for the user to allow methods to be called while building a multipart/form-data request while these do not belong in the construction of one.